### PR TITLE
Port the README to do_parse!

### PR DESCRIPTION
**Pull request by Keruspe**
Fixes last part of #328

Last piece, the README update, to delay until the release not to confuse users of nom 1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -308,15 +308,15 @@ fn ret_int1(i:&[u8]) -> IResult<&[u8], u8> { Done(i,1) }
...
-    tag!("abcd")? ~      // the question mark indicates an optional parser
-    aa: ret_int1  ~      // the return value of ret_int1, if it does not fail, will be stored in aa
-    tag!("efgh")  ~
-    bb: ret_int2  ~
-    tag!("efgh")  ,      // end the chain with a comma
-
-    ||{A{a: aa, b: bb}} // the final closure will be able to use the variable defined previously
+  do_parse!(    // the parser takes a byte array as input, and returns an A struct
+    tag!("abcd")       >>      // begins with "abcd"
+    opt!(tag!("abcd")) >>      // the question mark indicates an optional parser
```
Comment (SnirkImmington) [inline README.md]: I think you left the comments in from the original version.

Comment (Keruspe) [inline README.md]: Indeed, fixing

File: README.md [hunk 2]
```diff
@@ -308,15 +308,15 @@ fn ret_int1(i:&[u8]) -> IResult<&[u8], u8> { Done(i,1) }
...
-    tag!("efgh")  ,      // end the chain with a comma
-
-    ||{A{a: aa, b: bb}} // the final closure will be able to use the variable defined previously
+  do_parse!(    // the parser takes a byte array as input, and returns an A struct
+    tag!("abcd")       >>      // begins with "abcd"
+    opt!(tag!("abcd")) >>      // this is an optional parser
+    aa: ret_int1       >>      // the return value of ret_int1, if it does not fail, will be stored in aa
+    tag!("efgh")       >>
+    bb: ret_int2       >>
+    tag!("efgh")       >>      // end the chain with a comma
```
Comment (Hywan) [inline README.md]: Where is the comma here :-)?

Comment (Keruspe) [inline README.md]: Indeed, this comment is useless now, fixed

**Comment by Geal** [comment]
Thanks!

**Pull request merged**